net/test/ut/client_server.c

```diff
@@ -401,6 +402,20 @@ void m0_net_test_client_server_bulk_ut(void)
 			       64, 0x100000,
 			       8, 16, 0x1000, 0x10000);
 }
+void m0_net_test_xprt_dymanic_reg_dereg_ut(void)
+{
+	print_xprt();
+	m0_mem_xprt_fini();
+	m0_mem_xprt_init();
+	print_xprt();
```

Comment:
print_xprt() will print something. But it does not check anything.
Developers  must look into the tracing logs and manually make sure the result is success.

This is wrong.

Here in UT, we need to check that:
after m0_mem_xprt_fini(), the "m0_net_bulk_mem_xprt" is not in the global array.
If it is default, the deault_xprt should be NULL.

After m0_mem_xprt_init(), the "m0_net_bulk_mem_xprt" is again in the global array.

Reply:
incorporated